gnovm/pkg/gnolang/preprocess.go

```diff
@@ -3607,10 +3616,12 @@ func doConvertType(store Store, last BlockNode, x *Expr, t Type) {
 //	case 2: isNamedConversion is called within evaluating make() or new()
 //	(uverse functions). It returns TypType (generic) which does have IsNamed appropriate
 func isNamedConversion(xt, t Type) bool {
-	if t == nil {
-		t = xt
+	// this happens when processing blank identifiers (_) in function parameters
+	// or when types are not fully resolved during preprocessing.
+	// relative test: gnovm/tests/files/blankidentifier7.gno
+	if xt == nil || t == nil {
+		return false
```

Comment:
please revert this as no sufficient reason for the modification. @notJoon

Reply:
It would be good to merge after this.

Reply:
In the current modification, a panic is occurred in `blankidentifier7` test if I remove this nil check. The panic occurs in `isNamedConversion()` due to a nil pointer derefernce when attempting to call `IsNamed()` on a nil type.

I think this happens during type conversion preprocessing where either `x` and `xt` parameters are nil.

In particular, if `x` or `xt` is nil, this condition tries to call `nil.IsNamed()`, which causes the nil pointer problem.
https://github.com/gnolang/gno/blob/2093d8a43c1ec632707f28021860ac52f5a80c26/gnovm/pkg/gnolang/preprocess.go#L3638-L3646

So the possible solutions are:

1. add `nil` check condition (current version):

```go
func isNamedConversion(xt, t Type) bool {
    if xt == nil || t == nil { // <-
        return false
    }
    
    // ...

    if !c1 && !c2 {
        if t.IsNamed() && !xt.IsNamed() ||
            !t.IsNamed() && xt.IsNamed() {
            return true
        }
    }
    return false
}
```

2. Check `nil` on calling side:

```go
func convertType(store Store, last BlockNode, x *Expr, t Type) {
    if t == nil {
        return // <-
    }
    // ...
}
```

I'm not sure why `x` or `xt` becomes `nil` when the parameter is named blank identifier, but here's my guess.

```go
func Foo(_ string) string { /* ... */ }
```

1. The blank identifier typically indicates to the compiler that a parameter won't be used within the function.
2. However, this becomes problematic because:
    - `isNamedConversion()` is called during the preprocess stage
    - At this stage, the compiler is processing type information
    - It appears that type information for parameters with this identifier is either not properly transmitted or omitted

These are the causes and solutions I came up with. Please let me know if theres other solutions or if I'm wrong. thank you!

Reply:
not necessary after discussion with @notJoon